src/journal/journal-file.c

```diff
@@ -1896,27 +1892,20 @@ static int journal_file_link_entry(JournalFile *f, Object *o, uint64_t offset) {
         return 0;
 }
 
-static int journal_file_append_entry_internal(
+static int journal_file_append_entry_object(
```

Comment:
To gather the creat and the init of the entry object with `f->header->n_entries` incrementation.

Reply:
i don't grok this? you say "gather", but you split a function in two tha tonly has a single, caller. doesn't make sense to me? what am I missing?

Reply:
"gather" meant: the creation of the object, the initialization of the objects and initialization of the related data in the header are updated all in a common function. Even if there's one caller I thought it would be a good idea to encapsulate all these operations into a single function.

Currently the increment of `f->header->n_entries` is done somewhere else meaning that if something wrong happens between the allocation of the object and `f->header->n_entries++` then the counter in the header will be out of sync.